### DISCUSSION TITLE
Automatically mark an attrribute as private

### DISCUSSION THREAD

**Discussion by MVrachev** [pydantic/pydantic#2471] [Feature Request]
Hi, I am wondering why there is the need to explicitly mark certain fields as `PrivateAttr` when
it's forbidden to use underscore as a prefix for a normal field name: 
https://github.com/samuelcolvin/pydantic/issues/1476#issuecomment-623986595

Can't `pydantic` automatically mark the attributes starting with an underscore as `PrivateAttr` without the need of specifically marking them as private?
This way users will easily name their attributes without wondering what names are allowed and when to use `PrivateAttr`.

**Comment by PrettyWood** [comment]
Hi @MVrachev
Did you check `Config.underscore_attrs_are_private`?

```py
from pydantic import BaseModel


class M(BaseModel):
    x: str
    y: int
    _private_x: str
    _private_y: int = 2

    class Config:
        underscore_attrs_are_private = True


m = M(x='q', y=1)
assert not hasattr(m, '_private_x')
m._private_x = 'w'
assert hasattr(m, '_private_x')
assert m._private_x == 'w'
assert m._private_y == 2
```

Hope it helps!

**Comment by MVrachev** [comment]
Yes, but this still requires me to create a `Config` class and use the `underscore_attrs_are_private` field.
**Why do I have to use anything at all?**

Why not enable this by default when you **can't have** a public attribute with a name starting with an underscore?

**Comment by samuelcolvin** [comment]
To avoid backwards compatibility problem in minor releases.

I guess we could change it in v2. Any reason not to @PrettyWood?

**Reply by PrettyWood** [reply]
Yep was thinking the same! It would make sense

**Reply by MVrachev** [reply]
That would be awesome!
It would make it so much easier to interact with private attributes.
